src/discussions/posts/data/slices.js

```diff
@@ -32,13 +48,20 @@ const threadsSlice = createSlice({
     totalPages: null,
     totalThreads: null,
     postStatus: RequestStatus.SUCCESSFUL,
+    sortedBy: ThreadOrdering.BY_LAST_ACTIVITY,
+    statusFilter: PostsStatusFilter.ALL,
+    allTypeFilter: AllPostsFilter.ALL_POSTS,
+    myTypeFilter: MyPostsFilter.MY_POSTS,
+    search: '',
   },
   reducers: {
     fetchThreadsRequest: (state) => {
       state.status = RequestStatus.IN_PROGRESS;
     },
     fetchThreadsSuccess: (state, { payload }) => {
       state.status = RequestStatus.SUCCESSFUL;
+      state.topicThreadMap = {};
+      state.threads = {};
       normaliseThreads(state, payload.results);
```

Comment:
>  I would normalize before putting data into the reducer. It **should not** be done in the reducer, imho

Could you elaborate?

We're building up the state based on API responses from multiple different APIs, and they are all inter-related, and the API code doesn't really know the broader state it needs to act on. So, the reducer seemed like the most natural place since it has all the data. 

I can definitely move it if that's preferable. 

>  Also, the reducer is manipulating state directly, which is a redux anti-pattern. It should create a New state object. Manipulating state in place like that can cause all sorts of issues.

The state is **not** being manipulated in place. I've been using React and Redux long enough to know not to do that :-)
What's happening here is that this project, and other Open edX frontends use a library called [redux toolkit](https://redux-toolkit.js.org/), which is where the [createSlice](https://redux-toolkit.js.org/api/createSlice) function is coming from. This library automatically creates a new state object for you so you can manipulate with comfort and the state remains immutable.

You can also see it in action [here for instance](https://github.com/edx/frontend-app-learning/blob/master/src/course-home/data/slice.js)

Reply:
Thanks for the context regarding redux toolkit.

Please do not normalize the data in the reducer, the reducer is only suppose to update state not change / update any given state. If you can update, that would be great. thank you.

Reply:
Sure, will do. It's a biggish change so won't be able to get to it today.